examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,155 @@
+# Generate synthetic dataset
+# --------------------------
+from sklearn.datasets import make_regression
+
+X, y, w = make_regression(
+    n_samples=100,
+    n_features=100,
+    n_informative=10,
+    noise=8,
+    coef=True,
+    random_state=42,
+)
 
 # %%
-# Fit the ARD Regression
-clf = ARDRegression(compute_score=True)
-clf.fit(X, y)
-
-ols = LinearRegression()
-ols.fit(X, y)
+# Fit the regressions
+# -------------------
+import pandas as pd
+from sklearn.linear_model import ARDRegression, LinearRegression, BayesianRidge
+
+olr = LinearRegression()
+brr = BayesianRidge(compute_score=True, n_iter=30)
+ard = ARDRegression(compute_score=True, n_iter=30)
+
+olr.fit(X, y)
+brr.fit(X, y)
+ard.fit(X, y)
+
+df = pd.DataFrame(
+    {
+        "LinearRegression": olr.coef_,
+        "BayesianRidge": brr.coef_,
+        "ARDRegression": ard.coef_,
+        "ground truth": w,
+    }
+)
 
 # %%
-# Plot the true weights, the estimated weights, the histogram of the
-# weights, and predictions with standard deviations
-plt.figure(figsize=(6, 5))
-plt.title("Weights of the model")
-plt.plot(clf.coef_, color="darkblue", linestyle="-", linewidth=2, label="ARD estimate")
-plt.plot(
-    ols.coef_, color="yellowgreen", linestyle=":", linewidth=2, label="OLS estimate"
-)
-plt.plot(w, color="orange", linestyle="-", linewidth=2, label="Ground truth")
-plt.xlabel("Features")
-plt.ylabel("Values of the weights")
-plt.legend(loc=1)
-
-plt.figure(figsize=(6, 5))
-plt.title("Histogram of the weights")
-plt.hist(clf.coef_, bins=n_features, color="navy", log=True)
-plt.scatter(
-    clf.coef_[relevant_features],
-    np.full(len(relevant_features), 5.0),
-    color="gold",
-    marker="o",
-    label="Relevant features",
+# Plot the true and estimated coefficients
+# ----------------------------------------
+import matplotlib.pyplot as plt
+import seaborn as sns
+from matplotlib.colors import LogNorm
+
+plt.figure(figsize=(10, 6))
+ax = sns.heatmap(
+    df.T,
+    norm=LogNorm(),
+    cbar_kws={"label": "coefficients' values"},
+    cmap="viridis_r",
 )
-plt.ylabel("Features")
-plt.xlabel("Values of the weights")
-plt.legend(loc=1)
+plt.ylabel("linear model")
+plt.xlabel("coefficients")
+plt.tight_layout(rect=(0, 0, 1, 0.95))
+_ = plt.title("Models' coefficients")
 
-plt.figure(figsize=(6, 5))
-plt.title("Marginal log-likelihood")
-plt.plot(clf.scores_, color="navy", linewidth=2)
+# %%
+# Compared to the OLS (ordinary least squares) estimator, the coefficients using
+# a Bayesian Ridge regression are slightly shifted toward zeros, which
+# stabilises them. The ARD regression sets some of the non-informative
+# coefficients exactly to zero, while shifting some them closer to zero. Some
+# non-informative coefficients are still present and retain the order of
+# magnitude of the ground truth coefficients.
+
+# %%
+# Plot the marginal log-likelihood
+# --------------------------------
+plt.plot(ard.scores_, color="navy", label="ARD with polynomial features")
```

Comment:
If I am not wrong, we don't have a polynomial preprocessor for the data at this stage.